telebot/__init__.py

```diff
@@ -2598,6 +2598,10 @@ def send_document(
             logger.warning('The parameter "thumb" is deprecated. Use "thumbnail" instead.')
             thumbnail = thumb
 
+        if isinstance(document, types.InputFile) and visible_file_name:
+            # inputfile name ignored, warn
+            logger.warning('Cannot use both InputFile and visible_file_name. InputFile name will be ignored.')
```

Comment:
`files[key] = (value.file_name, value.file)`

visible_file_name will be ignoread, isn't it?

Reply:
No, it returns a tuple and in apihelper, tuple value[0] is unchanged, thus ignoring the inputfile naming